internal/xds/translator/translator.go

```diff
@@ -7,17 +7,19 @@ import (
 	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
 	resource "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/go-logr/logr"
 	"github.com/tetratelabs/multierror"
 
 	"github.com/envoyproxy/gateway/internal/ir"
 	"github.com/envoyproxy/gateway/internal/xds/types"
 )
 
 // Translate translates the XDS IR into xDS resources
-func Translate(ir *ir.Xds) (*types.ResourceVersionTable, error) {
+func Translate(log logr.Logger, ir *ir.Xds) (*types.ResourceVersionTable, error) {
```

Comment:
instead of changing the API, the IRs (xds and infra) could be printed as a yaml string in the gateway api runner

Reply:
I created #344 so we can make a decision on how to pass a logger.

Reply:
@danehans thanks for raising https://github.com/envoyproxy/gateway/issues/344 but its not related to my comment. I dont think we should be logging in libraries, instead we should we logging inside the clients consuming the libraries, in this case its the gateway-api runner (similar to what is done here https://github.com/envoyproxy/gateway/pull/302/commits/991f12caad34c6812188e1c85fe04aef8780b2d5)